### PR TITLE
Add implementation for custom filter `shodan/custom`

### PR DESCRIPTION
**Pull request by hyche**
Also bump version to 1.26

### CODE REVIEW

File: shodan/__main__.py [hunk 1]
```diff
@@ -617,22 +617,23 @@ def stats(limit, facets, filename, query):
 
 
 @main.command()
-@click.option('--color/--no-color', default=True)
```
Comment (sebix) [inline shodan/__main__.py:620]: Why did you rearrange all these lines? With all these changes it's much harder to compare/review the PR.

File: shodan/__main__.py [hunk 2]
```diff
@@ -617,22 +617,23 @@ def stats(limit, facets, filename, query):
...
-@click.option('--compresslevel', help='The gzip compression level (0-9; 0 = no compression, 9 = most compression', default=9, type=int)
 @click.option('--vulns', help='A comma-separated list of vulnerabilities to grab data on.', default=None, type=str)
-def stream(color, fields, separator, limit, datadir, ports, quiet, timeout, streamer, countries, asn, alert, tags, compresslevel, vulns):
+@click.option('--limit', help='The number of results you want to download. -1 to download all the data possible.', default=-1, type=int)
+@click.option('--compresslevel', help='The gzip compression level (0-9; 0 = no compression, 9 = most compression', default=9, type=int)
+@click.option('--timeout', help='Timeout. Should the shodan stream cease to send data, then timeout after <timeout> seconds.', default=0, type=int)
+@click.option('--color/--no-color', default=True)
+@click.option('--quiet', help='Disable the printing of information to the screen.', is_flag=True)
+def stream(streamer, fields, separator, datadir, asn, alert, countries, custom, ports, tags, vulns, limit, compresslevel, timeout, color, quiet):
```
Comment (sebix) [inline shodan/__main__.py:636]: Why did you change the order of parameters here? That's a major API change.

File: shodan/__main__.py [hunk 3]
```diff
@@ -617,22 +617,23 @@ def stats(limit, facets, filename, query):
...
-@click.option('--ports', help='A comma-separated list of ports to grab data on.', default=None, type=str)
-@click.option('--quiet', help='Disable the printing of information to the screen.', is_flag=True)
-@click.option('--timeout', help='Timeout. Should the shodan stream cease to send data, then timeout after <timeout> seconds.', default=0, type=int)
-@click.option('--streamer', help='Specify a custom Shodan stream server to use for grabbing data.', default='https://stream.shodan.io', type=str)
-@click.option('--countries', help='A comma-separated list of countries to grab data on.', default=None, type=str)
 @click.option('--asn', help='A comma-separated list of ASNs to grab data on.', default=None, type=str)
 @click.option('--alert', help='The network alert ID or "all" to subscribe to all network alerts on your account.', default=None, type=str)
+@click.option('--countries', help='A comma-separated list of countries to grab data on.', default=None, type=str)
+@click.option('--custom', help='A space-separated list of filters query to grab data on.', default=None, type=str)
```
Comment (sebix) [inline shodan/__main__.py:627]: I propose to use another parameter name than "custom", e.g. "custom-filter"
```suggestion
@click.option('--custom-filter', help='A space-separated list of filters query to grab data on.', default=None, type=str)
```

File: shodan/stream.py [hunk 4]
```diff
@@ -27,9 +27,12 @@ def _create_stream(self, name, timeout=None):
         # If the user requested a timeout then we need to disable heartbeat messages
         # which are intended to keep stream connections alive even if there isn't any data
         # flowing through.
-        if timeout:
+        if timeout is not None:
```
Comment (sebix) [inline shodan/stream.py:30]: That changes the behaviour if `timeout = False` in an unintuitive way.

**Review by sebix — CHANGES_REQUESTED** [review]

File: shodan/__main__.py (hunk 2 above)
Comment (hyche) [reply shodan/__main__.py:636]: Re-arrange the order of parameters here is according to rearranging the lines above, this is because I find the help section of `shodan stream` is not clear enough, so I group them into specific section. For example, you now can see `asn, alert, countries, custom, ports, tags, and vulns` are now next to each other, which probably make them more intuitive in my opinion. There is no API break here unless you import and call `stream()` directly, but even so, append new `custom` parameter also breaks the API.

File: shodan/__main__.py (hunk 3 above)
Comment (hyche) [reply shodan/__main__.py:627]: Agree, but I will change it to plural noun instead so `--custom-filters`, since we already have `countries`, or `ports`

File: shodan/stream.py (hunk 4 above)
Comment (hyche) [reply shodan/stream.py:30]: Compare to None should be `is` or `is not` according to PEP 8 best practice, this value should be either `None` or a number, it shouldn't be a boolean value

Comment (achillean) [reply shodan/stream.py:30]: This is a potentially breaking change. I agree that it would be better if we made it clear that it should be ``None`` or a number but that's not how it works at the moment. I think there's a conversation to be had about creating a Python3 version that uses type hints and properly conforms to PEP8 though that's outside the scope of what this MR is about.

File: shodan/stream.py [hunk 5]
```diff
@@ -27,9 +27,12 @@ def _create_stream(self, name, timeout=None):
...
             params['heartbeat'] = False
 
+        if query is not None:
```
Comment (achillean) [inline shodan/stream.py:33]: It feels a bit weird to handle this in ``_create_stream()`` instead of the ``custom()`` method as the ``query`` parameter is only used for custom streams. It might be nicer if we modify ``_create_stream()`` to have a ``**kwargs`` parameter whose key/ values get put into the ``params`` object.

Comment (achillean) [reply shodan/stream.py:33]: The downside of that is some kwargs would be treated differently than others then (ex. ``timeout`` wouldn't be put into the ``params`` object).

File: shodan/__main__.py (hunk 1 above)
Comment (achillean) [reply shodan/__main__.py:620]: Agreed. Please put these sorts of changes in a separate PR or add a comment in the PR what changed in the available options.
